node/src/components/sync_leaper.rs

```diff
@@ -268,6 +268,29 @@ impl SyncLeaper {
         }
     }
 
+    #[cfg_attr(doc, aquamarine::aquamarine)]
+    /// ```mermaid
+    /// flowchart TD
+    ///     style Start fill:#66ccff,stroke:#333,stroke-width:4px
+    ///     style End fill:#66ccff,stroke:#333,stroke-width:4px
+    ///
+    ///     title[SyncLeap process - AttemptLeap]
+    ///     title---Start
+    ///     style title fill:#FFF,stroke:#FFF
+    ///     linkStyle 0 stroke-width:0;
+    ///
+    ///     Start --> A{have at least<br>one peer?}
+    ///     A -->|Yes| B{is other sync<br>leap in progress?}
+    ///     A -->|No| End
+    ///     B -->|Yes| C{do sync leap<br>identifiers match?}
+    ///     C -->|No| End
+    ///     C -->|Yes| D[fetch SyncLeap from potentially<br>newly learned peers]
+    ///     B -->|No| G[fetch SyncLeap<br>from all peers]
+    ///     G --> E
+    ///     D -.- E[SyncLeap arrives]
```

Comment:
Did you mean for this to be a dotted line?

Reply:
That's a remnant from one of my previous approaches - I changed it to a normal arrow as it should be right now.

09ca60ebb01c4151d3d19c75ec6b9715ca663d81